HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,55 @@
+from pyspark import SparkContext
+from pyspark.accumulators import AccumulatorParam
+
+class VectorAccumulatorParam(AccumulatorParam):
+    def zero(self, value):
+        return [0.0] * len(value)
+    def addInPlace(self, val1, val2):
+        for i in xrange(len(val1)):
+             val1[i] += val2[i]
+        return val1
+
+# Parallel breadth first search
+def bfsParallel(graphRDD, character, sc, log, numNodesVisited, maxDiameter):
+	filterRDD = graphRDD.filter(lambda x: x[0] == character)
+
+	# Set used to record the visited nodes.
```

Comment:
Maybe we can use rdd to record visited nodes... I am not sure which performs better.